tls/s2n_resume.c

```diff
@@ -202,7 +202,8 @@ static S2N_RESULT s2n_tls13_client_deserialize_session_state(struct s2n_connecti
     uint8_t secret[S2N_TLS_SECRET_LEN] = { 0 };
     RESULT_GUARD_POSIX(s2n_stuffer_read_bytes(from, secret, secret_len));
 
-    /* Remove all previously-set PSKs */
+    /* Remove all previously-set resumption PSKs */
```

Comment:
It doesn't look like this removes any existing PSKs, just resets the type?

Reply:
The `s2n_psk_params_wipe` call removes ALL existing PSKs.

We have to do the type check before we do the wipe to make sure we're not switching from external->resumption. Do you think I could rephrase the comment to make that clearer?